contracts/managers/SocialTradingManager.sol

```diff
@@ -0,0 +1,296 @@
+ * on the RebalancingSetTokenV2 must be called through the administrative functions exposed on this contract.
+ */
+contract SocialTradingManager {
+    using SafeMath for uint256;
+
+    /* ============ Structs ============ */
+    struct PoolInfo {
+        address trader;
+        ISocialAllocator allocator;
+        uint256 currentAllocation;
+    }
+
+    /* ============ Events ============ */
+
+    event TradingPoolCreated(
+        address indexed _trader,
+        ISocialAllocator indexed _allocator,
+        address _tradingPool,
+        uint256 _startingAllocation
+    );
+
+    event AllocationUpdate(
+        address indexed _tradingPool,
+        uint256 _oldAllocation,
+        uint256 _newAllocation
+    );
+
+    event NewTrader(
+        address indexed _tradingPool,
+        address _oldTrader,
+        address _newTrader
+    );
+
+    /* ============ Modifier ============ */
+
+    modifier onlyTrader(IRebalancingSetTokenV2 _tradingPool) {
+        require(
+            msg.sender == pools[address(_tradingPool)].trader,
+            "Sender must be trader"
+        );
+        _;
+    }
+
+    /* ============ Constants ============ */
+
+    uint256 public constant REBALANCING_SET_NATURAL_UNIT = 10 ** 6;
+    uint256 public constant SCALE_FACTOR = 10 ** 18;
+
+    /* ============ State Variables ============ */
+
+    ICore public core;
+    mapping(address => PoolInfo) public pools;
+
+    /*
+     * SocialTradingManager constructor.
+     *
+     * @param  _core                            The address of the Core contract
+     */
+    constructor(
+        ICore _core
+    )
+        public
+    {
+        core = _core;
+    }
+
+    /* ============ External ============ */
+
+    /*
+     * Create a trading pool. Create or select new collateral and create RebalancingSetToken contract to
+     * administer pool. Save relevant data to pool's entry in pools state variable under the Rebalancing
+     * Set Token address.
+     *
+     * @param _tradingPairAllocator             The address of the allocator the trader wishes to use
+     * @param _factory                          Factory to use for RebalancingSetToken creation
+     * @param _startingBaseAssetAllocation      Starting base asset allocation in a scaled decimal value
+     *                                          (e.g. 100% = 1e18, 1% = 1e16)
+     * @param _startingValue                    Starting value of one share of the trading pool to 18 decimals of precision
+     * @param _name                             The name of the new RebalancingSetTokenV2
+     * @param _symbol                           The symbol of the new RebalancingSetTokenV2
+     * @param _rebalancingSetCallData           Byte string containing additional call parameters to pass to factory
+     */
+    function createTradingPool(
+        ISocialAllocator _tradingPairAllocator,
+        address _factory,
```

Comment:
Why choose not to hardcode factory hard-coded in the state of this contract?

Reply:
Potential for different RebalancingSet variations